src/main/java/com/algorithmia/algorithmHandler/RequestHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.algorithmia.algorithmHandler;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.Scanner;
+import org.apache.commons.codec.binary.Base64;
+
+
+final class RequestHandler<INPUT>
+{
+
+    private Scanner input;
+    private JsonParser parser;
+
+    RequestHandler(){
+        this.input = new Scanner(System.in);
+        this.parser = new JsonParser();
```

Comment:
Is JsonParser stateful or not thread-safe? I'm wondering why you've declared it as an instance variable rather than a class-level variable.